### PR TITLE
incremental: add cases for toc change and reference change

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs [hunk 1]
```diff
@@ -2382,6 +2378,632 @@ public void TestManagedReferenceDependencyRemove()
...
+                        new Dictionary<string, object>
+                        {
+                            ["meta"] = "Hello world!",
+                        },
+                        templateFolder: templateFolder,
+                        intermediateFolder: intermediateFolder);
+
+                }
+
+                ClearListener();
```
Comment (hellosnow) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs]: >ClearListener [](start = 16, length = 13)

minor: may consider to reuse IncrementalActions in IncrementalTestBase

**Comment by hellosnow** [comment]
:shipit:

File: test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs (hunk 1 above)
Comment (ansyral) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs]: would change it altogether later.

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**